src/main/scala/com/snowflake/snowpark/types/Variant.scala

```diff
@@ -164,7 +164,11 @@ class Variant private[snowpark] (
   def this(str: String) =
     this({
       try {
-        MAPPER.readTree(str)
+        if (str.toLowerCase().startsWith("null") && str != "null") {
```

Comment:
Let's add a comment here. For example, 
```
// `ObjectMapper` only reads the first token from the input string but not the whole string.
// For example, it can successfully convert "null dummy" to `null` value, but not reports error. 
```

Reply:
Got it!